File: 0215-kth-largest-element-in-an-array/0215-kth-largest-element-in-an-array.cpp
class Solution {
public:
    int findKthLargest(vector<int>& nums, int k) {
        // the greater<int> turns the heap from max_heap into min_heap
        // the greater<int> can be modified into self-defined cmp
        priority_queue<int, vector<int>, greater<int>> heap;

        for (int i=0; i<k; i++)
        {
            heap.push(nums[i]);
        }
        for (int i=k; i<nums.size(); i++)
        {
            heap.push(nums[i]);
            heap.pop();
        }

        return heap.top();
    }
};

File: 0215-kth-largest-element-in-an-array/0215-kth-largest-element-in-an-array.py
class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        max_heap = nums[0:k]

        heapq.heapify(max_heap)
        for x in range(k, len(nums)):
            # if the value is greater than the min in heap
            # pop the min out and push the curr value in
            heapq.heappushpop(max_heap, nums[x])
        
        return heapq.heappop(max_heap)